src/shell/layer/mod.rs

```diff
@@ -348,19 +346,19 @@ bitflags! {
     }
 }
 
-/// The configure state of a layer
+/// A layer surface configure.
 ///
-/// This type indicates compositor changes to the layer, such as a new size.
+/// A configure describes a compositor request to resize the layer surface or change it's state.
 #[non_exhaustive]
 #[derive(Debug, Clone)]
 pub struct LayerSurfaceConfigure {
-    /// The compositor suggested new size of the layer.
+    /// The compositor suggested new size of the layer in surface-local coordinates.
     ///
-    /// The size is a hint, meaning the client is free to ignore the new size (if the client does not resize),
-    /// pick a smaller size to satisfy aspect ratio or resize in steps. If you pick a small size and the
-    /// surface is anchored to two opposite anchors, then surface will be centered on the axis.
+    /// The size is a hint, meaning the new size can be ignored. A smaller size could be picked to satisfy
+    /// some aspect ratio or resize in steps. If the size is smaller than suggested and the layer surface is
+    /// anchored to two opposite anchors then the layer surface will be centered on that axis.
     ///
-    /// If either the width or height is 0, the compositor may choose any size for that specific width or height.
+    /// If either the width or height is 0, any size dimensions may be used.
```

Comment:
Something is missing here, the intent is that is the width is 0, you may choose any width you want, and same for the height.

Reply:
Would using `Option<NonZeroU32>` instead of `u32` help to show the intent? Also this would prevent some mistakes. By the way, having two named fields instead of a tuple might help too.

Reply:
I'd like to be consistent across shells regarding the fields if possible.

I plan to add a `SurfaceConstraints` type in a later pull request. This type would provide some helpers to correctly set the size of the surface (considering things such as max width/height, configure bounds (applicable in xdg shell) and scale (I think scale would involve a way to scale some constraints.